### PR TITLE
Shows routes

### PR DESCRIPTION
**Pull request by abachant**
Add basic routing for shows

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
   "dependencies": {
     "axios": "^0.21.1",
     "babel-loader": "^8.2.2",
+    "body-parser": "^1.19.0",
```
Comment (zkodjo) [inline package.json]: Remove body-parser as it's deprecated. Instead use express.json() as middleware when needed. or express.urlencoded()
Additional configs: https://expressjs.com/en/api.html#express.json

File: server/models/show.js [hunk 2]
```diff
@@ -18,13 +18,15 @@ module.exports = (sequelize, DataTypes) => {
   };
   Show.init({
     name: DataTypes.STRING,
```
Comment (zkodjo) [inline server/models/show.js:20]: Did you add a migration for this? Will be needed to update the DB. Also, I tried to say this yesterday to the whole group, I made a number of edits to the models that might affect the schemas but I think all those changes got incorporated.

File: server/models/show.js [hunk 3]
```diff
@@ -18,13 +18,15 @@ module.exports = (sequelize, DataTypes) => {
...
+    bandName: DataTypes.TEXT,
+    displayName: DataTypes.TEXT,
     showDate: DataTypes.DATE,
     latitude: DataTypes.FLOAT,
     longitude: DataTypes.FLOAT,
     city: DataTypes.STRING(64),
     state: DataTypes.STRING(64),
     country: DataTypes.STRING(64),
-    BandId: {
+    bandId: {
```
Comment (zkodjo) [inline server/models/show.js]: Please leave this as TitleCase - the field in the database is in title case

File: server/routes/shows.js [hunk 4]
```diff
@@ -1,4 +1,23 @@
-// const express = require('express');
+const express = require('express');
 // const logger = require('../../config/winston');
+const bodyParser = require('body-parser')
```
Comment (zkodjo) [inline server/routes/shows.js]: See previous comment

File: server/controllers/shows.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+const { Show } = require('../models');
+
+const logger = require('../../config/winston');
+
+exports.getShows = (req, res, next) => Show.findAll({
+  attributes: { exclude: ['BandId', 'createdAt', 'updatedAt'] },
+})
+  .then((result) => {
+    res.send(result);
```
Comment (zkodjo) [inline server/controllers/shows.js]: Can you add add a conditional to handle empty query results to tell the client they are a 404 - Not Found error? Sung found this bug yesterday, if a query can't find a result but the query is formed correctly it just returns null and it won't throw an error.

File: server/controllers/shows.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+exports.getBandShows = (req, res, next) => {
+  const { bandID } = req.params;
+  return Show.findAll({
+    attributes: { exclude: ['BandId', 'createdAt', 'updatedAt'] },
+    where: {
+      bandId: bandID,
+    },
+  })
+    .then((result) => {
+      res.send(result);
```
Comment (zkodjo) [inline server/controllers/shows.js]: Same as above - send back 404 on empty result

**Review by zkodjo — CHANGES_REQUESTED** [review]
Hi @abachant I left some comments inline

**Review by abachant — COMMENTED** [review]
Sounds good, I will adjust

**Review by abachant — COMMENTED** [review]
Has been updated

**Review by abachant — COMMENTED** [review]
Changed

**Review by zkodjo — APPROVED** [review]
LGTM, thanks

**Pull request merged**